packages/std/src/memory.rs

```diff
@@ -69,15 +69,84 @@ pub unsafe fn consume_region(ptr: *mut Region) -> Vec<u8> {
     )
 }
 
+/// Element that can be used to construct a new `Box<Region>`
+///
+/// # Safety
+///
+/// The following invariant must be upheld:
+///
+/// - full allocated capacity == value returned by capacity
+///
+/// This is important to uphold the safety invariant of the `dealloc` method, which requires us to pass the same Layout
+/// into it as was used to allocate a memory region.
+///
+/// And since `size` is one of the parameters, it is important to pass in the exact same capacity.
```

Comment:
```suggestion
/// And since `size` is one of the parameters, it is important to pass in the exact same capacity.
```

It was not obvious to me that size belongs to Layout. I think we can just put it in the same paragraph to glue the two together.